burn-book/src/saving-and-loading.md

```diff
@@ -0,0 +1,89 @@
+
+```rust, ignore
+// Save model in MessagePack format with full precision
+let recorder = NamedMpkFileRecorder::<FullPrecisionSettings>::new();
+model
+  .save_file(model_path, &recorder)
+  .expect("Could not save model");
+```
+
+Note that the file extension is automatically handled by the recorder depending on the one you choose. Therefore, only the file path and base name can be provided.
+
+Now that you have a trained model saved to your disk, you can easily load it in a similar fashion.
+
+```rust, ignore
+// Load model in full precision from MessagePack file
+let recorder = NamedMpkFileRecorder::<FullPrecisionSettings>::new();
+model
+  .load_file(model_path, &recorder)
+  .expect("Failed to load model file");
+```
+
+**Note:** models can be saved in different output formats, just make sure you are using the correct recorder type when loading the saved model. Type conversion between different precision settings is automatically handled, but formats are not interchangeable. A model can be loaded from one format and saved to another format, just as long as you load it back with the new recorder type afterwards.
+
+## Custom Initialization from Recorded Weights
+
+While the first approach is very straightforward, it does require the model to already be initialized. If instead you would like to skip the initialization and directly load the weights into the modules of your model, you can create a new initialization function. Let's take the following model definition as a simple example.
+
+```rust, ignore
+#[derive(Module, Debug)]
+pub struct Model<B: Backend> {
+    linear_in: Linear<B>,
+    linear_out: Linear<B>,
+    activation: ReLU,
+}
+```
+
+Similar to the [basic workflow inference example](../basic-workflow/inference.md), we can define a new initialization function which initializes the different parts of our model with the record values.
+
+```rust, ignore
+impl<B: Backend> Model<B> {
+    /// Returns the initialized model using the recorded weights.
+    pub fn init_with(record: ModelRecord<B>) -> Model<B> {
+        Model {
+            linear_in: LinearConfig::new(10, 64).init_with(record.linear_in),
+            linear_out: LinearConfig::new(64, 2).init_with(record.linear_out),
+            activation: ReLU::new(),
+        }
+    }
+
+    /// Returns the dummy model with randomly initialized weights.
+    pub fn new(device: &Device<B>) -> Model<B> {
+        let l1 = LinearConfig::new(10, 64).init(device);
+        let l2 = LinearConfig::new(64, 2).init(device);
+        Model {
+            linear_in: l1,
+            linear_out: l2,
+            activation: ReLU::new(),
+        }
+    }
+}
+```
+
+Now, let's save a model that we can load later. In the following snippets, we use `type Backend = NdArray<f32>` but you can use whatever backend you like.
+
+```rust, ignore
+// Create a dummy initialized model to save
+let model = Model::<Backend>::new(&Default::default());
```

Comment:
I would add an extra line `let device = Default::default();` just to be more specific.